massifs/logdircache.go

```diff
@@ -714,3 +727,15 @@ func dirFromFilepath(path string) (string, error) {
 	}
 	return path, nil
 }
+
+// checkLogPathAgainstCache checks the given cache log path against the given log Path to ensure they are the same log
+// from the same tenant.
+func checkLogPathAgainstCache(cacheLogPath string, logPath string) bool {
+
```

Comment:
This needs to deal with the case where the local reader is created with the "WithDirCacheTenant" options (case b) in my comments on the ticket). As logPath is the path being considered for reading, the current working dir must be appropriate OR logPath is already absoloute. Why not do cacheLogPath == os.Abs(logPath) ?

You just need to check that cacheLogPath is always stored as the Abs of what is read into the cache

Reply:
I feel suffix matching is probably better? then it just takes where the replica is stored/was stored locally out of the equation